src/ZPublisher/request_params.rst

```diff
@@ -0,0 +1,296 @@
+request's ``Content-Type``), but they, too, have a *name* and a *value*.
+
+All request parameter names and values are strings.
+A parameter value, however, often designates a value of a specific type,
+e.g. an integer or a datetime. The response generating code often can
+be significantly simplified when it does not need to make the
+type conversion itself. In addition, in some cases, the request parameters
+are not independent from one another, but some of them are related; in those
+cases, it can help, when the related parameters
+are aggregated into a single object. Zope supports both cases but it needs
+directives to guide the process. It uses *name* suffixes of the form
+``:``\ *directive* to specify such directives. For example,
+the parameter ``i:int=1`` tells Zope to convert the value ``'1'`` to an
+integer and use it as value for request variable ``i``; the parameter sequence
+``x.name:record=Peter&x.age:int:record=10`` tells Zope to contruct
+a record ``x`` with attributes ``name`` and ``age`` and respective values ``'Peter'`` and ``10``.
+
+The remaining parts of this document describe the available kinds
+of directives and the overall processing model.
+
+
+Request parameter directive kinds
+=================================
+
+Zope supports different kinds of request parameter directives.
+This sections describes them in detail.
+
+
+Converter
+---------
+
+A converter directive converts the parameter value (a string) to a converter
+specific type in a coverter specific way. The available converters are given
+by ``ZPublisher.Converters.type_converters``.
+
+Simple type converters are ``int``, ``float``, ``boolean``, ...
+each converting to the respective type.
+
+Some converters employ special logic for the conversion.
+For example, both ``tokens`` as well as ``lines`` convert to
+a list of strings but ``tokens`` splits the input at whitespace, ``lines``
+at newlines.
+
+The converter ``required`` contains only logic: it checks that
+the value is not empty.
+
+Different Zope versions can support different converters.
+
+
+Encoding
+--------
+
+An encoding directive tells the converting process the encoding
+of the parameter value. Typical encodings are "utf8", "latin1" ...
+
+An encoding directive is ignored, if the parameter does not
+have a converter directive as well. In some sense, the encoding
+directive informs the converter about the value's encoding.
+If there is no encoding directive, the converter uses the
+default encoding (specified via the configuration option
+``zpublisher-default-encoding``).
+
+In principle, Zope supports any encoding known by the ``codecs``
+module. However, the converter may impose restrictions.
+
+
+Aggregator
+----------
+
+An aggregator directive tells Zope how to process parameters with
+a similar name.
+
+Zope supports the following aggregators:
+
+list
+  collect all values with this name into a list.
+
+  If there are more than a single parameter with the same name,
```

Comment:
"If there is..." - but I am not 100% sure about this.

Reply:
I have reformulated to ensure the plural is correct.